controllers/integration_test.go

```diff
@@ -538,6 +538,109 @@ var _ = Context("INTEGRATION: Inside of a new namespace", func() {
 			}
 		})
 
+		It("should create and scale organization's repository runners on workflow_job event", func() {
+			name := "example-runnerdeploy"
+
+			{
+				rd := &actionsv1alpha1.RunnerDeployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: ns.Name,
+					},
+					Spec: actionsv1alpha1.RunnerDeploymentSpec{
+						Replicas: intPtr(1),
+						Selector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"foo": "bar",
+							},
+						},
+						Template: actionsv1alpha1.RunnerTemplate{
+							ObjectMeta: metav1.ObjectMeta{
+								Labels: map[string]string{
+									"foo": "bar",
+								},
+							},
+							Spec: actionsv1alpha1.RunnerSpec{
+								RunnerConfig: actionsv1alpha1.RunnerConfig{
+									Repository: "test/valid",
+									Image:      "bar",
+									Group:      "baz",
+								},
+								RunnerPodSpec: actionsv1alpha1.RunnerPodSpec{
+									Env: []corev1.EnvVar{
+										{Name: "FOO", Value: "FOOVALUE"},
+									},
+								},
+							},
+						},
+					},
+				}
+
+				ExpectCreate(ctx, rd, "test RunnerDeployment")
+				ExpectRunnerSetsCountEventuallyEquals(ctx, ns.Name, 1)
+				ExpectRunnerSetsManagedReplicasCountEventuallyEquals(ctx, ns.Name, 1)
+				env.ExpectRegisteredNumberCountEventuallyEquals(1, "count of fake list runners")
+			}
+
+			// Scale-up to 1 replica via ScaleUpTriggers.GitHubEvent.WorkflowJob based scaling
+			{
+				hra := &actionsv1alpha1.HorizontalRunnerAutoscaler{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: ns.Name,
+					},
+					Spec: actionsv1alpha1.HorizontalRunnerAutoscalerSpec{
+						ScaleTargetRef: actionsv1alpha1.ScaleTargetRef{
+							Name: name,
+						},
+						MinReplicas:                       intPtr(1),
+						MaxReplicas:                       intPtr(5),
+						ScaleDownDelaySecondsAfterScaleUp: intPtr(1),
+						ScaleUpTriggers: []actionsv1alpha1.ScaleUpTrigger{
+							{
+								GitHubEvent: &actionsv1alpha1.GitHubEventScaleUpTriggerSpec{
+									WorkflowJob: &actionsv1alpha1.WorkflowJobSpec{
+										Status:       "queued",
```

Comment:
Hey! Thanks for the PR. This looks good overall, but could you remove every fields under `WorkflowJobSpec`?

The whole point of `workflow_job` for me is that it frees us from manually mapping `status` and other fields to a subset of runners to be scaled.

For example, we currently map every `workflow_job` of `action=queued` to an internal `scale by 1` event, while mapping one with `action=completed` to an internal `scale down by 1` event. So, manually specifying this defeats our original motivation on using `workflow_job`, probably.

I thought it was the same for other fields like `repositories`, `names`, and `labels`. But I'd need to recheck 🤔

Reply:
> The whole point of workflow_job for me is that it frees us from manually mapping status and other fields to a subset of runners to be scaled.

I can imagine situations where users might want more control over scaling behavior in response to `WorkflowJob` webhooks. For example: scaling up only when the workflow name matches a pattern, which would require `WorkflowJobSpec.Names`. 

Is your intention that users could accomplish that use-case by different means, or is it not a use-case you want to support?

(Thanks for the review, Yusuke!)

Reply:
long time no speak 😅 , could you remove the attributes so we can merge this? We want to have the event to be explicit as implemented via this PR with the plan to remove the default event entirely however we unsure about the `WorkflowJob` attributes so would like to get this merged without them first and go from there.

Reply:
Will do! Thanks for the reminder, @toast-gear

Reply:
@toast-gear: removed all `WorkflowJobSpec` fields in 1002e8f7e10b52dc32eff5ab6b5885fddcc1fb28. Now it's just an empty struct type.